pycir/cir.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+"""
+Centered isotonic regression, generic but more tailored to dose-response and dose-finding data
+
+Assaf Oron, recoded from R 'cir' package
+"""
+
+import numpy as np
+import pandas as pd
+
+
+def cirPAVA(y, x=None, wt=None, outx=None, interiorStrict=True, strict=False, ybounds=None, full=False, dec=False):
+    """
+    Perform regression.
+
+    Args:
+        x,y,wt: vectors of equal length with the doses, mean y values (usually response rates in [0,1]), and weights (often sample size)
+        outx: ...
+    """
+    if ybounds is None:
+        ybounds = np.asarray([0.0,1.0])
```

Comment:
One of the biggest "gotchas" with python is that if you define a mutable object such as a list, dictionary, or array in the function signature, then it will use the same object for all instances of the function, which you almost never want. The workaround is to define it as None and then actually assign it in the function body.